Added support for RegExp in allowOrigins

dauer opened the pull request:
If setting 'allowOriginsRegExp' to true in the config it is possible to
specify RegExp rules in the 'allowOrign' array instead of urls.

dauer committed:
Added support for RegExp in allowOrigins

If setting 'allowOriginsRegExp' to true in the config it is possible to
specify RegExp rules in the 'allowOrign' array instead of urls.

davidolrik commented on src/CorsService.php:
```diff
@@ -152,6 +154,10 @@ protected function configure(array $config)
                 }
+                // Build a single regexp for matching
+                if ( isset($config['allowOriginsRegExp']) && $config['allowOriginsRegExp'] ) {
+                    $this->allowOriginsRegExp = "/^(" . implode('|', $this->allowOrigins) . ")$/";
```
Should the regex not be changed as follows?

```diff
- $this->allowOriginsRegExp = "/^(" . implode('|', $this->allowOrigins) . ")$/";
+ $this->allowOriginsRegExp = "/^(?:" . implode('|', $this->allowOrigins) . ")$/";
```

(To avoid the capture of the match)

dauer committed:
Review: RegExp no longer catches the match

davidolrik reviewed (approved):
🖖🏼

dauer committed:
Added method for collecting CORS headers on Request object

Adding CORS headers to the HTTP response is the last thing we do in a
long chain of commands. If one fails (throws an exception) we risk not having the CORS headers
added. This ensures we have access to the relevant headers even if an error
should occure.

polesen commented on src/CorsService.php:
```diff
@@ -90,25 +90,44 @@ public function handlePreflightRequest(Request $request)
+     */
+    public function handleRequest(Request $request, Response $response)
+    {
+        if (empty($request->attributes->get('x-sentry-cors-headers'))) {
```
why this? Du kalder den jo allerede i middleware?

dauer committed:
Removed call to corsHeaders() in handleRequest()

The call was only there to please the test, so I moved it to the test
code...

dauer committed:
Cleanup outcommented code
